tests/integration/test_charm.py

```diff
@@ -198,3 +201,123 @@ async def test_scale_down_capablities(ops_test: OpsTest) -> None:
+    ]
+    password = await get_password(ops_test, app=ANOTHER_DATABASE_APP_NAME)
+    client = replica_set_client(ip_addresses, password, app=ANOTHER_DATABASE_APP_NAME)
+
+    db = client["new-db"]
+    test_collection = db["test_collection"]
+    ubuntu = {"release_name": "Jammy Jelly", "version": 22.04, "LTS": False}
+    test_collection.insert(ubuntu)
+
+    # read all entries from new cluster
+    cursor = test_collection.find({})
+    cluster_1_entries = set()
+    for document in cursor:
+        logger.error(document)
+        cluster_1_entries.add(document["release_name"])
+
+    client.close()
+
+    # read all entries from other cluster
+    ip_addresses = [unit.public_address for unit in ops_test.model.applications[APP_NAME].units]
+    password = await get_password(ops_test)
+    client = replica_set_client(ip_addresses, password)
+
+    db = client["new-db"]
+    test_collection = db["test_collection"]
+
+    # read all entries from original cluster
+    cursor = test_collection.find({})
+    cluster_2_entries = set()
+    for document in cursor:
+        cluster_2_entries.add(document["release_name"])
+
+    client.close()
+
+    common_entries = cluster_2_entries.intersection(cluster_1_entries)
+    logger.error(cluster_1_entries)
+    logger.error(cluster_2_entries)
+    assert len(common_entries) == 0, "Writes from one cluster are replicated to another cluster."
+
+
+async def test_replication_member_scaling(ops_test: OpsTest) -> None:
+    """Verify newly added and newly removed members properly replica data.
+
+    Verify newly members have replicated data and newly removed members are gone without data.
+    """
+    original_ip_addresses = [
+        unit.public_address for unit in ops_test.model.applications[APP_NAME].units
+    ]
+    await ops_test.model.applications[APP_NAME].add_unit(count=1)
+    await ops_test.model.wait_for_idle(apps=[APP_NAME], status="active", timeout=1000)
+    assert len(ops_test.model.applications[APP_NAME].units) == 4
+
+    new_ip_addresses = [
+        unit.public_address for unit in ops_test.model.applications[APP_NAME].units
+    ]
+    new_member_ip = list(set(new_ip_addresses) - set(original_ip_addresses))[0]
+    password = await get_password(ops_test)
+    client = MongoClient(unit_uri(new_member_ip, password), directConnection=True)
+
+    # check for replicated data while retrying to give time for replica to copy over data.
+    try:
+        for attempt in Retrying(stop=stop_after_delay(2 * 60), wait=wait_fixed(3)):
+            with attempt:
+                db = client["new-db"]
+                test_collection = db["test_collection"]
+                query = test_collection.find({}, {"release_name": 1})
+                logger.error(query[0]["release_name"])
+                assert query[0]["release_name"] == "Focal Fossa"
+
+    except RetryError:
+        assert False, "Newly added unit doesn't replicate data."
```

Comment:
Use `with pytest.raises(RetryError):` here. 
If you need to close the client connection, `finally: client.close()`

Reply:
Try not to test two things at once, try to split them up if needed.